src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -348,6 +348,23 @@ public function findOrNew($id, $columns = ['*'])
     }
 
     /**
+     * Determine if a record exists by its primary key.
+     *
+     * @param  mixed  $id
+     * @return bool
+     */
+    public function exists($id = null)
+    {
+        // if an array of ids is passed, this returns true
+        // when any one of the ids exist.
+        if (! is_null($id)) {
+            $this->whereKey($id);
+        }
+
```

Comment:
You could use a `->unless()` here.

Reply:
So this instead:

```
return $this->unless(is_null($id), function ($builder, $value) use ($id) {
    return $builder->whereKey($id);                                       
})->query->exists();                                                      
```

Reply:
Yeah, just an idea.

Reply:
It is more fluent, but IMO less readable.  It just moves the if into the unless.  

Happy to change the PR to use the above code.  Let me know which you would prefer.